onnxruntime/core/providers/cpu/ml/treeregressor.cc

```diff
@@ -55,8 +74,17 @@ common::Status TreeEnsembleRegressor<T>::Compute(OpKernelContext* context) const
   }
   int64_t N = X->Shape().NumDimensions() == 1 ? 1 : X->Shape()[0];
   Tensor* Y = context->Output(0, {N, tree_ensemble_.n_targets_or_classes_});
-
-  tree_ensemble_.compute(context, X, Y, NULL);
+  if (std::is_same<TH, TO>::value) {
+    TH* y_data = Y->template MutableData<TH>();
+    tree_ensemble_.compute(context, X, y_data, NULL);
+  } else {
+    std::vector<TH> y_data_th(Y->Shape().Size());
```

Comment:
>Y->Shape().Size()); [](http://example.com/codeflow?start=30&length=19)

reserve() + push_back is a better option

Reply:
I need to create a buffer. It is filled in another function. Most of the time it is parallelized. push_back is not possible.

Reply:
It is the initialization that is killing us and shows up on the profiler. If zeros are not necessary, would std::unique_ptr<TH[]> be a better option?

Reply:
Sorry, I forgot std::vector was initializing every elements. I changed into a pointer.